src/Microsoft.Azure.SignalR.AspNet/OwinExtensions.cs

```diff
@@ -273,5 +277,94 @@ private static IReadOnlyList<string> GetAvailableHubNames(HubConfiguration confi
             var hubManager = configuration.Resolver.Resolve<IHubManager>();
             return hubManager?.GetHubs().Select(s => s.Name).ToList();
         }
+
+        /// <summary>
+        /// From https://github.com/Azure/azure-signalr/blob/dev/src/Microsoft.Azure.SignalR.AspNet/OwinExtensions.cs
+        /// TODO: merge with the one in SignalR when it is refactored
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="applicationName"></param>
+        /// <param name="configuration"></param>
+        /// <param name="options"></param>
+        /// <returns></returns>
+        private static IAppBuilder InitAzureSignalRHost(this IAppBuilder builder, string applicationName, HubConfiguration configuration, ServiceOptions options)
+        {
+            // 1. Init SignalR Host 
+            EnsureValidCulture();
+
+            // Ensure we have the conversions for MS.Owin so that
+            // the app builder respects the OwinMiddleware base class
+            SignatureConversions.AddConversions(builder);
+
+            var env = builder.Properties;
+
+            // If we don't get a valid instance name then generate a random one
+            string instanceName = env.GetAppInstanceName() ?? Guid.NewGuid().ToString();
+
+            // always use empty ProtectedData
+            var protectedData = new EmptyProtectedData();
```

Comment:
Do we still need this? We added an override to PersistentConnection to allow overriding this right? https://github.com/SignalR/SignalR/issues/4175